<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

In [1]:
# Exercise 1 (Learning to interrupt)
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

Exercise 2 (Resetting & Restarting the Kernal)

In [3]:
print = 'PRINT'
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello World')

Hello World


Exercise 3 (Timing Stuff)

In [3]:
%timeit squares = [i**2 for i in range(10)]

1.32 μs ± 92.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [5]:
squares = [i**2 for i in range(10)]
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [6]:
squares = list()
for i in range(10):
    squares.append(i**2)
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [4]:
%%timeit 
squares = list()
for i in range(10):
    squares.append(i**2)

1.71 μs ± 285 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


1.32 is the mean time taken to run each loop while 92.1 is the standard deviation of it. There are 7 runs total and each run consist of 1 million loops.

1.71 is the mean time taken to run each loop while 285 is the standard deviation of it. There are 7 runs total and each run consist of 1 million loops.

The first code is faster.